src/java.base/share/classes/javax/crypto/Cipher.java

```diff
@@ -1049,12 +1049,13 @@ public final byte[] getIV() {
      * Returns the parameters used with this cipher.
      *
      * <p>The returned parameters may be the same that were used to initialize
-     * this cipher, or may contain a combination of default and random
-     * parameter values used by the underlying cipher implementation if this
-     * cipher requires algorithm parameters but was not initialized with any.
+     * this cipher, or may contain additional default or random parameter
+     * values used by the underlying cipher implementation. If the required
```

Comment:
The PR for https://github.com/openjdk/jdk/pull/8396/ has one difference in this sentence in that it says "... values used by the underlying signature implementation _if the underlying signature implementation supports returning the parameters as {@code AlgorithmParameters}_." Should this also be consistent?

Also, I don't think you need to repeat "underlying signature " again above, you could just say: ... values used by the underlying signature implementation _if the implementation supports returning the parameters as {@code AlgorithmParameters}_."

Reply:
Signature class has that sentence is due to the peculiar case of EdDSA signature impl where its parameter spec does not correlate to an ASN.1 structure. Since there is no such case for current cipher impls (as far as I know), that's why I have not added it to Cipher class. 

Ok, how about `If the required parameters were not supplied and can be generated by the cipher, the generated parameters will be returned.`?